wpimath/src/main/native/cpp/controller/PIDController.cpp

```diff
@@ -18,6 +18,24 @@ using namespace frc;
 PIDController::PIDController(double Kp, double Ki, double Kd,
                              units::second_t period)
     : m_Kp(Kp), m_Ki(Ki), m_Kd(Kd), m_period(period) {
+  if (Kp < 0.0) {
+    wpi::math::MathSharedStore::ReportError(
+        "Kp must be a non-negative number, got {}!", Kp);
+    m_Kp = 0.0;
+    wpi::math::MathSharedStore::ReportWarning("Kp defaulted to 0.");
```

Comment:
If we default Kp to 0, should it set the other terms to zero as well, to ensure the mechanism won't move?